### ISSUE TITLE
Zesty build failing...

### ISSUE THREAD

**Issue by Panoplos** [rethinkdb/rethinkdb#6377]
Tried to build according to README on Zesty, core dumped.

### Config
```fish
⟩ ./configure --allow-fetch --prefix=/opt
* Detecting system configuration
Bash:                           4.4.7(1)-release
Use ccache:                     no
C++ Compiler:                   GCC 6.3.0 (/usr/bin/c++)
Host System:                    x86_64-linux-gnu
Build System:                   Linux 4.10.0-21-generic x86_64
Cross-compiling:                no
Host Operating System:          Linux
Build Architecture:             x86_64
C++11:                          ok
Precompiled web assets:         no
Protobuf compiler:              /usr/bin/protoc
python:                         python 2.7.13
Node.js package manager:        /usr/bin/npm
coffee:                         external/coffee-script_1.10.0
Browserify:                     external/browserify_13.1.0
bluebird:                       external/bluebird_2.9.32
web UI dependencies:            external/admin-deps_2.0.4
wget:                           /usr/bin/wget
curl:                           /usr/bin/curl
Google Test:                    external/gtest_1.7.0
termcap:                        -ltermcap
boost_system:                   -lboost_system
protobuf:                       -lprotobuf
v8 javascript engine:           external/v8_3.30.33.16-patched
RE2:                            external/re2_2015-11-01
z:                              -lz
crypto:                         external/openssl_1.0.1u
ssl:                            external/openssl_1.0.1u
curl:                           -lcurl
malloc:                         jemalloc
jemalloc (static):              external/jemalloc_4.1.0
Test protobuf:                  ok
Test boost:                     ok
Installation prefix:            /opt
Configuration prefix:           /opt/etc
Runtime data prefix:            /opt/var
* Wrote configuration to config.mk
```
### Failure
```fish
⟩ make -j4
    [1/459] FETCH coffee-script_1.10.0
    [2/459] BUILD bluebird_2.9.32
    [3/459] FETCH browserify_13.1.0
    [4/459] FETCH admin-deps_2.0.4
    [5/459] INSTALL-INCLUDE gtest_1.7.0
    [6/459] FETCH re2_2015-11-01
    [7/459] FETCH v8_3.30.33.16-patched
    [8/459] FETCH openssl_1.0.1u
    [9/459] FETCH jemalloc_4.1.0
    ...
  /usr/bin/c++ -pthread -m64 -m64  -o /home/makoto/Downloads/rethinkdb/build/external/v8_3.30.33.16-patched/build/out/x64.release/mksnapshot -Wl,--start-group /home/makoto/Downloads/rethinkdb/build/external/v8_3.30.33.16-patched/build/out/x64.release/obj.target/mksnapshot/src/mksnapshot.o /home/makoto/Downloads/rethinkdb/build/external/v8_3.30.33.16-patched/build/out/x64.release/obj.target/tools/gyp/libv8_base.a /home/makoto/Downloads/rethinkdb/build/external/v8_3.30.33.16-patched/build/out/x64.release/obj.target/tools/gyp/libv8_nosnapshot.a /home/makoto/Downloads/rethinkdb/build/external/v8_3.30.33.16-patched/build/out/x64.release/obj.target/tools/gyp/libv8_libplatform.a /home/makoto/Downloads/rethinkdb/build/external/v8_3.30.33.16-patched/build/out/x64.release/obj.target/third_party/icu/libicui18n.a /home/makoto/Downloads/rethinkdb/build/external/v8_3.30.33.16-patched/build/out/x64.release/obj.target/third_party/icu/libicuuc.a /home/makoto/Downloads/rethinkdb/build/external/v8_3.30.33.16-patched/build/out/x64.release/obj.target/tools/gyp/libv8_libbase.a /home/makoto/Downloads/rethinkdb/build/external/v8_3.30.33.16-patched/build/out/x64.release/obj.target/third_party/icu/libicudata.a -Wl,--end-group -lrt
  LD_LIBRARY_PATH=/home/makoto/Downloads/rethinkdb/build/external/v8_3.30.33.16-patched/build/out/x64.release/lib.host:/home/makoto/Downloads/rethinkdb/build/external/v8_3.30.33.16-patched/build/out/x64.release/lib.target:$LD_LIBRARY_PATH; export LD_LIBRARY_PATH; cd ../tools/gyp; mkdir -p /home/makoto/Downloads/rethinkdb/build/external/v8_3.30.33.16-patched/build/out/x64.release/obj.target/v8_snapshot/geni; "/home/makoto/Downloads/rethinkdb/build/external/v8_3.30.33.16-patched/build/out/x64.release/mksnapshot" --log-snapshot-positions --logfile "/home/makoto/Downloads/rethinkdb/build/external/v8_3.30.33.16-patched/build/out/x64.release/obj.target/v8_snapshot/geni/snapshot.log" --random-seed 314159265 "/home/makoto/Downloads/rethinkdb/build/external/v8_3.30.33.16-patched/build/out/x64.release/obj.target/v8_snapshot/geni/snapshot.cc"
Segmentation fault (core dumped)
tools/gyp/v8_snapshot.target.x64.release.mk:13: recipe for target '/home/makoto/Downloads/rethinkdb/build/external/v8_3.30.33.16-patched/build/out/x64.release/obj.target/v8_snapshot/geni/snapshot.cc' failed
make[3]: *** [/home/makoto/Downloads/rethinkdb/build/external/v8_3.30.33.16-patched/build/out/x64.release/obj.target/v8_snapshot/geni/snapshot.cc] Error 139
make[3]: Leaving directory '/home/makoto/Downloads/rethinkdb/build/external/v8_3.30.33.16-patched/build/out'
Makefile:286: recipe for target 'x64.release' failed
make[2]: *** [x64.release] Error 2
make[2]: Leaving directory '/home/makoto/Downloads/rethinkdb/build/external/v8_3.30.33.16-patched/build'

Full error log: ./build/external/v8_3.30.33.16-patched_install.log
mk/support/build.mk:148: recipe for target 'build/external/v8_3.30.33.16-patched/lib/libv8.a' failed
make[1]: *** [build/external/v8_3.30.33.16-patched/lib/libv8.a] Error 1
make[1]: *** Waiting for unfinished jobs....
Makefile:52: recipe for target 'make' failed
make: *** [make] Error 2
```
Also, any reason this build is pulling down a universe of libraries and building when I have all requisites on the computer?

**Comment by srh** [comment]
Try adding CXX=g++-5 to your ./configure line.  And if that doesn't work, adding `CXXFLAGS="-fno-delete-null-pointer-checks"` on the configure line will make it work -- I have that on the last configure I did on my machine, but I forget if that really needs to be there.

What versions of libraries do you have on your computer that RethinkDB shouldn't be downloading?

**Comment by Panoplos** [comment]
OK, tested with:

```fish
⟩ ./configure --allow-fetch --prefix=/opt CXX=g++-5
* Detecting system configuration
Bash:                           4.4.7(1)-release
Use ccache:                     no
C++ Compiler:                   GCC 5.4.1 (g++-5)
Host System:                    x86_64-linux-gnu
Build System:                   Linux 4.10.0-21-generic x86_64
Cross-compiling:                no
Host Operating System:          Linux
Build Architecture:             x86_64
C++11:                          ok
Precompiled web assets:         no
Protobuf compiler:              /usr/bin/protoc
python:                         python 2.7.13
Node.js package manager:        /usr/bin/npm
coffee:                         external/coffee-script_1.10.0
Browserify:                     external/browserify_13.1.0
bluebird:                       external/bluebird_2.9.32
web UI dependencies:            external/admin-deps_2.0.4
wget:                           /usr/bin/wget
curl:                           /usr/bin/curl
Google Test:                    external/gtest_1.7.0
termcap:                        -ltermcap
boost_system:                   -lboost_system
protobuf:                       -lprotobuf
v8 javascript engine:           external/v8_3.30.33.16-patched
RE2:                            external/re2_2015-11-01
z:                              -lz
crypto:                         external/openssl_1.0.1u
ssl:                            external/openssl_1.0.1u
curl:                           -lcurl
malloc:                         jemalloc
jemalloc (static):              external/jemalloc_4.1.0
Test protobuf:                  ok
Test boost:                     ok
Installation prefix:            /opt
Configuration prefix:           /opt/etc
Runtime data prefix:            /opt/var
* Wrote configuration to config.mk
```
But now getting a linking error:
```fish
⟩ make -j4
    [1/421] CC build/release/obj/web_assets/web_assets.o
    [2/421] CC build/release/obj/rdb_protocol/ql2.pb.o
    [3/421] BUILD v8_3.30.33.16-patched
    [4/421] CC build/release/obj/activity_logger.o
    ...
    [421/421] LD build/release/rethinkdb
/usr/bin/ld: /home/makoto/Downloads/rethinkdb/build/external/openssl_1.0.1u/lib/libssl.a(ssl_algs.o): undefined reference to symbol 'EVP_dss1@@OPENSSL_1.0.0'
//lib/x86_64-linux-gnu/libcrypto.so.1.0.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
src/build.mk:337: recipe for target 'build/release/rethinkdb' failed
make[1]: *** [build/release/rethinkdb] Error 1
Makefile:52: recipe for target 'make' failed
make: *** [make] Error 2
```

**Comment by Panoplos** [comment]
OK, I got the build working by uninstalling libcurl4-openssl-dev and installing g++-5 and libssl-dev, then running `./configure --allow-fetch --prefix=/opt CXX=g++-5`.

However, now it appears that this build is out of date:
```fish
~/Downloads/rethinkdb
⟩ rethinkdb
Recursively removing directory /home/makoto/Downloads/rethinkdb/rethinkdb_data/tmp
Initializing directory /home/makoto/Downloads/rethinkdb/rethinkdb_data
Running rethinkdb 2.3.2-windows-beta-405-gb51d7c (GCC 5.4.1)...
Running on Linux 4.10.0-21-generic x86_64
Loading data from directory /home/makoto/Downloads/rethinkdb/rethinkdb_data
Listening for intracluster connections on port 29015
Listening for client driver connections on port 28015
Listening for administrative HTTP connections on port 8080
Listening on cluster addresses: 127.0.0.1, 127.0.1.1, ::1
Listening on driver addresses: 127.0.0.1, 127.0.1.1, ::1
Listening on http addresses: 127.0.0.1, 127.0.1.1, ::1
To fully expose RethinkDB on the network, bind to all addresses by running rethinkdb with the `--bind all` command line option.
Server ready, "DevMetal_a1l" 3f3597ad-d4ad-4074-b88e-d4fe809de960
A newer version of the RethinkDB server is available: 2.3.5. You can read the changelog at <https://github.com/rethinkdb/rethinkdb/releases>.
```
But this was pulled from `next`. Just in case, I tried:
```fish
~/Downloads/rethinkdb
⟩ git pull origin next
From https://github.com/rethinkdb/rethinkdb
 * branch                  next       -> FETCH_HEAD
Already up-to-date.
```
So, not sure 1) why it is saying this is a Windows build, and 2) why it is out of date by 3 versions.

**Comment by AtnNn** [comment]
Closing as duplicate of #5757

It looks like you've opened #6378 for those last two issues.
